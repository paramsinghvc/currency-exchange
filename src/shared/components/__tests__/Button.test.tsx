import React from "react";
import Button from "../Button";
import renderer from "react-test-renderer";

describe("Should render Button component properly", () => {
  it("should render a flat button", () => {
    const flatButton = renderer.create(
      <Button onClick={() => {}} label="EXCHANGE" fullWidth={false} variant="flat" rounded size="x-large" />
    );
    expect(flatButton.toJSON()).toMatchSnapshot();
  });

  it("should render a outlined button", () => {
    const button = renderer.create(<Button onClick={() => {}} label="EXCHANGE" fullWidth={false} variant="outlined" />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it("should render a full-width button", () => {
    const button = renderer.create(<Button onClick={() => {}} label="EXCHANGE" fullWidth />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it("should render a rounded button", () => {
    const button = renderer.create(<Button onClick={() => {}} label="EXCHANGE" rounded />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it("should render a x-large button", () => {
    const button = renderer.create(<Button onClick={() => {}} label="EXCHANGE" size="x-large" />);
    expect(button.toJSON()).toMatchSnapshot();
  });
});
