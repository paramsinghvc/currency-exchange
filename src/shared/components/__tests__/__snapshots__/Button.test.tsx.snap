// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Button component properly should render a flat button 1`] = `
<button
  className="css-k394ta"
  color="#7b68ee"
  label="EXCHANGE"
  onClick={[Function]}
  size="x-large"
>
  EXCHANGE
</button>
`;

exports[`Should render Button component properly should render a full-width button 1`] = `
<button
  className="css-1uq2sz1"
  color="#7b68ee"
  label="EXCHANGE"
  onClick={[Function]}
  size="large"
>
  EXCHANGE
</button>
`;

exports[`Should render Button component properly should render a outlined button 1`] = `
<button
  className="css-jtv4lb"
  color="#7b68ee"
  label="EXCHANGE"
  onClick={[Function]}
  size="large"
>
  EXCHANGE
</button>
`;

exports[`Should render Button component properly should render a rounded button 1`] = `
<button
  className="css-1v5kw01"
  color="#7b68ee"
  label="EXCHANGE"
  onClick={[Function]}
  size="large"
>
  EXCHANGE
</button>
`;

exports[`Should render Button component properly should render a x-large button 1`] = `
<button
  className="css-dmogtu"
  color="#7b68ee"
  label="EXCHANGE"
  onClick={[Function]}
  size="x-large"
>
  EXCHANGE
</button>
`;
